src/app/clients/form/formSections/NumberAdultsCard.tsx

```diff
@@ -32,35 +28,38 @@ const getQuantity = (input: string): number => {
     return -1;
 };
 
-const getNumberAdults = (
+const setNumberAdults = (
     fieldSetter: FieldSetter,
     errorSetter: ErrorSetter,
-    adults: Person[],
-    gender: Gender
+    adults: NumberAdultsByGender,
+    fieldKey: string
 ): SelectChangeEventHandler => {
     return (event) => {
         const input = event.target.value;
-        const newValue = adults;
-        const personIndex = newValue.findIndex((person) => person.gender === gender);
+        const newAdults = { ...adults, [fieldKey]: getQuantity(input) };
+        fieldSetter("adults", newAdults);
 
-        newValue[personIndex].quantity = getQuantity(input);
-        fieldSetter("adults", newValue);
-
-        const invalidAdultEntry = newValue.filter((value) => value.quantity === -1);
-        const nonZeroAdultEntry = newValue.filter((value) => value.quantity! > 0);
+        const invalidAdultEntry =
+            newAdults.numberFemales === -1 ||
+            newAdults.numberMales === -1 ||
+            newAdults.numberUnknownGender === -1;
```

Comment:
I know we set numbers to -1 if invalid if i scroll up, but is it better to check for less than 0 just to be safe?
```suggestion
            newAdults.numberFemales < 0 ||
            newAdults.numberMales < 0 ||
            newAdults.numberUnknownGender < 0;
```